### PR TITLE
C port of cowsay added

### PR DESCRIPTION
**Pull request by erkinalp**
A very simple C port of cowsay with two cows and line length limiting functionality.

### CODE REVIEW

File: Applications/util/cowsay.c [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+"        ||     ||\n";
+
+char cowstring_fuzix[]=
+"   //------x   U    U  ========7  ====T====  \\   /    \n"
+"   U           U    U        //       I       \\ /     \n"
+"   U=======x   U    U       //        I        X      \n"
+"   U           U    U      //         I         X     \n"
+"   U            \\  /     //           I       // \\    \n"
+"   U             \\/    4========  ====Y====  //   \\   \n"
+"                             by Alan Cox and others   \n";
```
Comment (erkinalp) [inline Applications/util/cowsay.c]: Someone can improve the ASCII art here

File: Applications/util/cowsay.c [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        putchar('\n');
+   }
+   putchar(' ');
+   putchar('\\');
+   for (size_t i=0;i<linelen+2;i++) if (i==5) putchar('v'); else putchar('-');
+   putchar('/');
+   putchar('\n');
+}
+
+int main (int argc,char *argv[]) {
```
Comment (erkinalp) [inline Applications/util/cowsay.c]: Argument vector needs to be writable. Otherwise too much copying would happen.

**Comment by EtchedPixels** [comment]
Would be useful to have the library bits. Been busy and cowsay isn't a priority item 8)

**Comment by tormodvolden** [comment]
cowsay is very welcome! However this code needs to be made more readable. Please use proper indenting and whitespace. I am not sure about the hack of splicing arguments by poking spaces over the string terminators... When I tried to build this, optopt was not defined. Finally I think I ran into bugs in the provided string functions, but I deliberately didn't look at them. If the license issue sustains, I can "clean room" implement new ones.

**Comment by erkinalp** [comment]
argv is laid out linearly on flat-memory systems. Cowsay port exploits this. However, it will segfault on non-flat argv allocations. optopt is from getopt, providing _STH_ in -o _STH_.

**Comment by tormodvolden** [comment]
optopt is not declared in Library/include/getopt.h nor mentioned in getopt.c

File: Applications/util/cowsay.c [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    }
+    for (optind++;optind<=argc;optind++) {
+        strcat(newargv,argv[optind]);
+        realloc(newargv,newarglen=strlen(newargv+2));
+        if (optind<argc) {
+            newargv[newarglen++]=' ';
+            newargv[newarglen]=0;
+        }
+    }
+    realloc(newargv,strlen(newargv));
```
Comment (erkinalp) [inline Applications/util/cowsay.c]: Targets with MMU can get into page fault just before realloc()

**Pull request closed without merging**